src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/DetectPreviewFeatureAnalyzer.cs

```diff
@@ -1081,5 +1054,44 @@ private static bool SymbolIsAnnotatedAsPreview(ISymbol symbol, ConcurrentDiction
 
             return existing.isPreview;
         }
+
+        protected void AnalyzeTypeSyntax(SyntaxNodeAnalysisContext context, SyntaxNode node, ConcurrentDictionary<ISymbol, (bool isPreview, string? message, string? url)> requiresPreviewFeaturesSymbols, Func<ISymbol, bool> symbolIsAnnotatedAsPreview)
+        {
+            var type = context.SemanticModel.GetTypeInfo(node, context.CancellationToken).Type;
+            if (type is not null && symbolIsAnnotatedAsPreview(type))
+            {
```

Comment:
This is the core of the refactoring. I'm moving reporting diagnostics to this method.

Here, I'm analyzing `NameSyntax`, then referring back to `SemanticModel` to get the symbol I'm dealing with, and determine if we have a special diagnostic for it.